client_it_test.go

```diff
@@ -140,6 +143,64 @@ func TestClientMemberEvents(t *testing.T) {
 	})
 }
 
+func calcPartitionID(ss *serialization.Service, key interface{}) (int32, error) {
+	kd, err := ss.ToData(key)
+	if err != nil {
+		return 0, err
+	}
+	return murmur.HashToIndex(kd.PartitionHash(), 271), nil
+}
+
+func TestClientEventHandlingOrder(t *testing.T) {
+	// Create custom cluster, and client from it
+	cls := it.StartNewClusterWithOptions("event-order-test-cluster", 15701, it.MemberCount())
+	defer cls.Shutdown()
+	conf := cls.DefaultConfig()
+	ctx := context.Background()
+	c := it.MustValue(hz.StartNewClientWithConfig(ctx, conf)).(*hz.Client)
+	defer c.Shutdown(ctx)
+	ss := it.MustValue(serialization.NewService(&conf.Serialization)).(*serialization.Service)
+	// Create test map
+	m := it.MustValue(c.GetMap(ctx, "TestClientEventHandlingOrder")).(*hz.Map)
+	var lc hz.MapEntryListenerConfig
+	lc.NotifyEntryAdded(true)
+	var (
+		// have 271 partitions by default, starting from 1
```

Comment:
Why it starts from 1?

Reply:
I assumed it at the beginning and it seems I was wrong. Thanks a lot for catching that! Fixed at https://github.com/hazelcast/hazelcast-go-client/pull/696/commits/6faa426bc2a39bd8dbb77be78fb1465fee2e194d